### PR TITLE
Added validation for route name for http triggers

### PR DESCRIPTION
**Pull request by gavin-aguiar**
## Description
Added validation for route name for http triggers

Fixes #

---

### PR information

- [x] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azure_functions_worker/functions.py [hunk 1]
```diff
@@ -78,6 +78,15 @@ def get_return_binding(binding_name: str,
...
         return return_binding_name
 
+    @staticmethod
+    def validate_binding_route(func_name: str, binding: BindingInfo,
+                               func_type: str):
+        if hasattr(binding, 'route') and binding.route.startswith(
+                '/') and func_type == 'function':
+            raise FunctionLoadError(
+                func_name,
+                f'Invalid route name. {binding.route}')
```
Comment (vrdmr) [inline azure_functions_worker/functions.py:88]: Can you be verbose here? And also an action plan for the user - what should they change to get out of this error?

File: azure_functions_worker/functions.py [hunk 2]
```diff
@@ -368,6 +378,8 @@ def add_indexed_function(self, function_id: str,
 
         bound_params = {}
         for binding in function.get_bindings():
+            self.validate_binding_route(func_name, binding, func_type)
```
Comment (vrdmr) [inline azure_functions_worker/functions.py:381]: This is only useful for http. right?

If yes, then add a validate_bindings call, which internally calls validate binding route and validate binding direction in it.

In future, we might want to validate other bindings as well and we don't want to make it convulated.

**Review by vrdmr — CHANGES_REQUESTED** [review]

File: tests/unittests/test_functions_registry.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    def setUp(self) -> None:
+        def dummy():
+            return "test"
+
+        self.dummy = dummy
+        self.func = Function(self.dummy, "test.py")
+        self.function_registry = functions.Registry()
+
+    def test_add_index_functions_invalid_route(self):
```
Comment (YunchuWang) [inline tests/unittests/test_functions_registry.py:22]: add a test for happy path or already exists?

File: azure_functions_worker/functions.py (hunk 1 above)
Comment (gavin-aguiar) [reply azure_functions_worker/functions.py]: Fixed. Added a more verbose message which should tell the user that the trigger name cannot start with /

File: azure_functions_worker/functions.py (hunk 2 above)
Comment (gavin-aguiar) [reply azure_functions_worker/functions.py]: Refactored.

File: tests/unittests/test_functions_registry.py (hunk 3 above)
Comment (gavin-aguiar) [reply tests/unittests/test_functions_registry.py]: Happy paths already exists in the form on e2e tests.

**Review by vrdmr — APPROVED** [review]

**Review by YunchuWang — APPROVED** [review]

**Pull request merged**